qualtran/bloqs/bookkeeping/allocate.py

```diff
@@ -69,7 +69,7 @@ def adjoint(self) -> 'Bloq':
         return Free(self.dtype, self.dirty)
 
     def on_classical_vals(self) -> Dict[str, int]:
-        return {'reg': 0}
+        return {'reg': self.dtype.from_bits([0] * self.dtype.num_qubits)}
```

Comment:
This is fine for now, but let's add this to the forthcoming issue to track "friction related to adding new data types that aren't integers". Specifically: having allocate always allocate a bunch of zero bits that get interpreted in whatever encoding the dtype specifies is a fine choice but it's a choice nonetheless that we should probably discuss in the broader context of datatypes and it should be documented somewhere.

Reply:
Added to https://github.com/quantumlib/Qualtran/issues/1437